pylint/checkers/similar.py

```diff
@@ -137,37 +465,77 @@ def _get_similarity_report(self, similarities: List[Tuple]) -> str:
-            index1 += skip
-
-    def _iter_sims(self):
+    def _find_common(
+        self, lineset1: "LineSet", lineset2: "LineSet"
+    ) -> Generator[CommonalitiesT, None, None]:
+        """
+        Find similarities in the two given linesets.
+
+        This the core of the algorithm.
+        The idea is to compute the hashes of a minimal number of successive lines of each lineset and then compare the hashes.
+        Every match of such comparison is stored in a dict that links the couple of starting indices in both linesets to
+        the couple of corresponding starting and ending lines in both files.
+        Last regroups all successive couples in a bigger one. It allows to take into account common chunk of lines that have more
+        than the minimal number of successive lines required.
+        """
+        hash_to_index_1: HashToIndex_T
+        hash_to_index_2: HashToIndex_T
+        index_to_lines_1: IndexToLines_T
+        index_to_lines_2: IndexToLines_T
+        hash_to_index_1, index_to_lines_1 = hash_lineset(lineset1, self.min_lines)
+        hash_to_index_2, index_to_lines_2 = hash_lineset(lineset2, self.min_lines)
+
+        hash_1: FrozenSet[LinesChunk] = frozenset(hash_to_index_1.keys())
+        hash_2: FrozenSet[LinesChunk] = frozenset(hash_to_index_2.keys())
+
+        common_hashes: Iterable[LinesChunk] = sorted(
+            hash_1 & hash_2, key=lambda m: hash_to_index_1[m][0]
+        )
+
+        # all_couples is a dict that links the couple of indices in both linesets that mark the beginning of
+        # successive common lines, to the corresponding starting and ending number lines in both files
+        all_couples: CplIndexToCplLines_T = {}
+
+        for c_hash in sorted(common_hashes, key=operator.attrgetter("_index")):
+            for indices_in_linesets in itertools.product(
+                hash_to_index_1[c_hash], hash_to_index_2[c_hash]
+            ):
+                index_1 = indices_in_linesets[0]
+                index_2 = indices_in_linesets[1]
+                all_couples[
+                    LineSetStartCouple(index_1, index_2)
+                ] = CplSuccessiveLinesLimits(
+                    copy.copy(index_to_lines_1[index_1]),
+                    copy.copy(index_to_lines_2[index_2]),
+                    effective_cmn_lines_nb=self.min_lines,
+                )
+
+        remove_successives(all_couples)
+
+        for cml_stripped_l, cmn_l in all_couples.items():
+            file_1_lines = cmn_l.first_file
+            file_2_lines = cmn_l.second_file
+            start_index_1 = cml_stripped_l.fst_lineset_index
+            start_index_2 = cml_stripped_l.snd_lineset_index
+            nb_common_lines = cmn_l.effective_cmn_lines_nb
+
+            if (
+                filter_noncode_lines(
+                    lineset1, start_index_1, lineset2, start_index_2, nb_common_lines
+                )
+                > self.min_lines
+            ):
```

Comment:
The formatting here is terrible, unfortunately. If nothing else helps, maybe create a temp var to store the result of the `filter_noncode_lines(...)` call right before doing the comparison.

Reply:
Much nicer like this! Thanks.